helm/qontract-reconcile/templates/template.yaml

```diff
@@ -127,8 +127,102 @@ objects:
                 key: log_group_name
           command: ["/bin/sh", "-c"]
           args:
+          # generate /etc/vector/vector.yaml
+          # and generate /fluentd/etc/fluent.conf
           - |
-            # generate fluent.conf
+            ### APPSRE-8096: Vector Section Begins
+            mkdir -p /etc/vector
+            cat > /etc/vector/vector.yaml <<EOF
+            sources:
+              qr:
+                type: file
+                include:
+                  - /fluentd/log/integration.log
+                read_from: beginning
+                data_dir: /fluentd/log
+
+            transforms:
+              ### De-spaminate
+              despam_events:
+                type: filter
+                inputs:
+                  - qr
+                condition:
+                  type: datadog_search
+                  source: -"using gql endpoint" AND -"Certificate did not match expected hostname"
+
+              ### Craft a JSON document that has escaped double quotes and changes the event from the 
+              ### Vector-default `message` to `text`, which is require by Slack and Google Chat.
+              rename_message_webhooks:
+                type: remap
+                inputs:
+                  - despam_events
+                source: |-
+                  .message = join!(
```

Comment:
took me a minute to realize what the `!` meant, for future reviewers: https://vector.dev/docs/reference/vrl/expressions/#function_call-grammar

Reply:
To summarize:

In VRL, Vector considers the `join` method to have the potential to be "failable." (I'm not quite sure _how_ it can fail, since it joins multiple strings into one, but there you go.) VRL also supports appending an `!` to certain methods to discard any possible errors ("failures"). It's equivalent to Golang's `result, _ := methodWithError()`.

Reply:
Gotcha, appreciate the explanation!